platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/ExportMigrationContext.java

```diff
@@ -0,0 +1,114 @@
+ * tracking exported migration entries for a given migratable while processing an export migration
+ * operation.
+ * <p>
+ * <b>
+ * This code is experimental. While this interface is functional
+ * and tested, it may change or be removed in a future version of the
+ * library.
+ * </b>
+ * </p>
+ */
+public interface ExportMigrationContext extends MigrationContext {
+    /**
+     * Creates or retrieves (if already created) a migration entry referenced from the specified system
+     * property to be exported by the corresponding migratable.
+     * <p>
+     * An error will automatically be recorded with the associated migration report if the system
+     * property is not defined or its value is blank.
+     *
+     * @param name the name of the system property referencing a migration entry to create or retrieve
+     * @return a new migration entry or the existing one if already created for the migration entry
+     * referenced from the specified system property value or empty if the property is not defined or
+     * its value is blank
+     * @throws IllegalArgumentException if <code>name</code> is <code>null</code>
+     */
+    public default Optional<ExportMigrationEntry> getSystemPropertyReferencedEntry(String name) {
+        return getSystemPropertyReferencedEntry(name, (r, v) -> true);
+    }
+
+    /**
+     * Creates or retrieves (if already created) a migration entry referenced from the specified system
+     * property to be exported by the corresponding migratable.
+     * <p>
+     * The provided predicate will be invoked with the associated migration report and the property
+     * value (may be <code>null</code> if not defined). In case the property is not defined or is
+     * blank, an error will be automatically recorded unless the predicate returns <code>false</code>.
+     * Returning <code>true</code> in such case will still not create a corresponding migration entry.
+     * In all other cases, no errors or warning will be generated if the predicate returns <code>false</code>
+     * so it is up to the predicate to record one if required.
+     *
+     * @param name      the name of the system property referencing a migration entry to create or retrieve
+     * @param validator a predicate to be invoked to validate the property value further which must
+     *                  return <code>true</code> to have a migration entry created and returned
+     * @return a new migration entry or the existing one if already created for the migration entry
+     * referenced from the specified system property value or empty if the property is not defined or
+     * is invalid
+     * @throws IllegalArgumentException if <code>name</code> or <code>validator</code> is <code>null</code>
+     */
+    public Optional<ExportMigrationEntry> getSystemPropertyReferencedEntry(String name,
+            BiPredicate<MigrationReport, String> validator);
```

Comment:
Agreed. If I was to make it optional here, then it would be defined as an Optional and not as a @Nullable. But I do not want it to be optional here.